modules/openapi/src/internals/postprocess/TaggedUnionTransformer.scala

```diff
@@ -30,21 +30,37 @@ object TaggedUnionTransformer extends IModelPostProcessor {
     def empty: TaggedUnionInfo = TaggedUnionInfo(Vector.empty, isTagged = true)
   }
 
+  private def getFieldsFromAllParentsTransitively(
+      target: Definition,
+      allDefinitions: Vector[Definition]
+  ): Vector[Field] = {
+    target match {
+      case s: Structure =>
+        val allParents = allDefinitions.filter(d => s.parents.contains(d.id))
+        allParents.flatMap { parent =>
+          getFieldsFromAllParentsTransitively(
+            parent,
+            allDefinitions
+          )
+        } ++ s.localFields
+      case _ => Vector.empty
+    }
+  }
+
   private def getInfoForTargets(
-      targets: Vector[Definition]
+      targets: Vector[Definition],
+      allDefinitions: Vector[Definition]
   ): TaggedUnionInfo = {
     targets.foldLeft(TaggedUnionInfo.empty) { (info, t) =>
-      t match {
-        case Structure(_, localFields, parents, _) =>
-          val singleLocal = localFields.size == 1
-          val allLocalRequired =
-            localFields.forall(_.hints.contains(Hint.Required))
-          val noParents = parents.isEmpty
-          if (singleLocal && allLocalRequired && noParents)
-            info.addFields(localFields)
-          else info.setNotTagged
-        case _ => info.setNotTagged
-      }
+      val allFields =
+        getFieldsFromAllParentsTransitively(t, allDefinitions)
+      val singleLocal = allFields.size == 1
+      val allLocalRequired =
+        allFields.forall(_.hints.contains(Hint.Required))
+      val noParents = true // parents.isEmpty
```

Comment:
is this comment a reminder or you're pointing out that if we're here it's because `parents.isEmpty` was already called prior to reaching this point?

Reply:
Ah thanks for pointing out, I meant to go back to that.. I will fix.

Reply:
Yeah so before that was in the algorithm for tagged unions, but I updated it so now it will check all parents and combine the fields from all parents in order to get the fields (to check if it is a single required field or not)